include/xtensor-io/xfile_array.hpp

```diff
@@ -10,6 +10,23 @@
 
 namespace xt
 {
+    struct xfile_dirty
+    {
+        bool data_dirty;
+        bool shape_dirty;
+
+        xfile_dirty(bool is_dirty=false)
```

Comment:
why not two arguments, one for `data_dirty` and one for `shape_dirty`?

Reply:
We don't need it, we want to initialize an `xfile_array` either as dirty or as not dirty.